Session reconnect gives black screen until screen changes

This is xrdp v0.9.16, xorgxrdp v*.16, Linux, Guacamole, Firefox client, with initial-program (single application, not desktop) set.

When reconnecting to a session by reloading the browser tab, while not moving the mouse the screen remains black. Only when the mouse is moved AND some GUI element is hovered that triggers a "screen change", for example an input element that changes its appearance on hover, or the mouse cursor changing to the text input cursor, the screen goes from black to showing the actual program. If no such alement exists, e.g. when some modal dialog is blocking the rest of the UI, the black screen cannot be cured. 

Can you reproduce this with a more simple setup? If not, where else would you report this?

Setup of server1:
ubuntu 18.04 <- x.org <- xorgxrdp <- xrdp <- guacamole

Client:
guacamole's html5-client

There is no VNC anywhere. I tried to login to RDP from server1 to server1 with remmina, but that didn't work.

/var/log/xrdp.log with "Xorg" selected in the ugly GUI at login:
```
[20210810-10:40:40] [INFO ] connecting to sesman ip 127.0.0.1 port 3350
[20210810-10:40:40] [INFO ] xrdp_wm_log_msg: sesman connect ok
[20210810-10:40:40] [INFO ] sesman connect ok
[20210810-10:40:40] [INFO ] sending login info to session manager, please wait...
[20210810-10:40:42] [INFO ] xrdp_wm_log_msg: login failed for display 0
[20210810-10:40:42] [INFO ] login failed for display 0
```
with "Xvnc" (does that even make sense?):
```
[20210810-10:43:31] [INFO ] connecting to sesman ip 127.0.0.1 port 3350
[20210810-10:43:31] [INFO ] xrdp_wm_log_msg: sesman connect ok
[20210810-10:43:31] [INFO ] sesman connect ok
[20210810-10:43:31] [INFO ] sending login info to session manager, please wait...
[20210810-10:43:33] [INFO ] xrdp_wm_log_msg: login failed for display 0
[20210810-10:43:33] [INFO ] login failed for display 0
```
with "neutrinordp-any":
```
[20210810-10:44:33] [ERROR] xrdp_wm_log_msg: error loading libxrdpneutrinordp.so specified in xrdp.ini, please add a valid entry like lib=libxrdp-vnc.so or similar
[20210810-10:44:33] [INFO ] error loading libxrdpneutrinordp.so specified in xrdp.ini, p
[20210810-10:44:33] [INFO ] lease add a valid entry like lib=libxrdp-vnc.so or similar
[20210810-10:44:33] [ERROR] Failure setting up module
```
You said without sesman, but looking at the logs, it seems involved. The last log suggests adding a lib= entry, however I wonder why guacamole doesn't need that.

--- a/module/rdpClientCon.c
+++ b/module/rdpClientCon.c
@@ -1026,6 +1026,8 @@ rdpClientConProcessMsgClientInfo(rdpPtr dev, rdpClientCon *clientCon)
         clientCon->shmemstatus = shmemstatus;
     }
 
+    rdpClientConAddDirtyScreen(dev, clientCon, 0, 0, clientCon->rdp_width, clientCon->rdp_height);
+
     return 0;
 }
 

